node/core/approval-voting/src/lib.rs

```diff
@@ -1049,26 +1064,23 @@ async fn handle_actions<Context>(
 }
 
 fn cores_to_candidate_indices(
+	core_indices: &AssignmentBitfield,
 	block_entry: &BlockEntry,
-	candidate_index: CandidateIndex,
-	cert: &AssignmentCertV2,
-) -> Vec<CandidateIndex> {
+) -> Result<AssignmentBitfield, AssignmentBitfieldError> {
 	let mut candidate_indices = Vec::new();
-	match &cert.kind {
-		AssignmentCertKindV2::RelayVRFModuloCompact { sample: _, core_indices } => {
-			for cert_core_index in core_indices {
-				if let Some(candidate_index) = block_entry
-					.candidates()
-					.iter()
-					.position(|(core_index, _)| core_index == cert_core_index)
-				{
-					candidate_indices.push(candidate_index as _)
-				}
-			}
-		},
-		_ => candidate_indices.push(candidate_index as _),
+
+	// Map from core index to candidate index.
```

Comment:
do we check the len of `candidate_indices` is the same as `core_indices.iter_ones().len()` or simply filter out garbage?

Reply:
This fn doesn't really check that, but it is checked when the assignment is imported.

Reply:
Actually this code is only for our own assignments. `candidates_indices` must be greater or equal to number of 1 bits in the produced `AssignmentBitfield`.

`compute_relay_vrf_modulo_assignments_v2` needs the `leaving_cores` parameter to  filters out invalid core assignments. These are also checked when importing the assignment.